src/PointwiseFunctions/AnalyticData/Xcts/BinaryWithGravitationalWaves.cpp

```diff
@@ -0,0 +1,756 @@
+            square(distance_left);
+      }
+    }
+  }
+}
+
+template <typename DataType>
+void BinaryWithGravitationalWavesVariables<DataType>::operator()(
+    gsl::not_null<tnsr::ijk<DataType, 3>*> deriv_3_distance_right,
+    gsl::not_null<Cache*> cache,
+    ::Tags::deriv<
+        ::Tags::deriv<::Tags::deriv<detail::Tags::DistanceRight<DataType>,
+                                    tmpl::size_t<Dim>, Frame::Inertial>,
+                      tmpl::size_t<Dim>, Frame::Inertial>,
+        tmpl::size_t<Dim>, Frame::Inertial> /*meta*/) const {
+  const auto& distance_right =
+      get(cache->get_var(*this, detail::Tags::DistanceRight<DataType>{}));
+  const auto& normal_right =
+      cache->get_var(*this, detail::Tags::NormalRight<DataType>{});
+  std::array<std::array<double, 3>, 3> delta{
+      {{{1., 0., 0.}}, {{0., 1., 0.}}, {{0., 0., 1.}}}};
+  for (size_t i = 0; i < 3; ++i) {
+    for (size_t j = 0; j < 3; ++j) {
+      for (size_t k = 0; k < 3; ++k) {
+        deriv_3_distance_right->get(i, j, k) =
+            (-normal_right.get(i) * delta.at(j).at(k) -
+             normal_right.get(j) * delta.at(i).at(k) -
+             normal_right.get(k) * delta.at(i).at(j) +
+             3 * normal_right.get(i) * normal_right.get(j) *
+                 normal_right.get(k)) /
+            square(distance_right);
+      }
+    }
+  }
+}
+
+template <typename DataType>
+void BinaryWithGravitationalWavesVariables<DataType>::operator()(
+    const gsl::not_null<tnsr::I<DataType, 3>*> normal_left,
+    const gsl::not_null<Cache*> cache,
+    detail::Tags::NormalLeft<DataType> /*meta*/) const {
+  const auto& distance_left =
+      get(cache->get_var(*this, detail::Tags::DistanceLeft<DataType>{}));
+  get<0>(*normal_left) = (get<0>(x) - xcoord_left) / distance_left;
+  get<1>(*normal_left) = get<1>(x) / distance_left;
+  get<2>(*normal_left) = get<2>(x) / distance_left;
+}
+
+template <typename DataType>
+void BinaryWithGravitationalWavesVariables<DataType>::operator()(
+    const gsl::not_null<tnsr::I<DataType, 3>*> normal_right,
+    const gsl::not_null<Cache*> cache,
+    detail::Tags::NormalRight<DataType> /*meta*/) const {
+  const auto& distance_right =
+      get(cache->get_var(*this, detail::Tags::DistanceRight<DataType>{}));
+  get<0>(*normal_right) = (get<0>(x) - xcoord_right) / distance_right;
+  get<1>(*normal_right) = get<1>(x) / distance_right;
+  get<2>(*normal_right) = get<2>(x) / distance_right;
+}
+
+template <typename DataType>
+void BinaryWithGravitationalWavesVariables<DataType>::operator()(
+    const gsl::not_null<tnsr::ii<DataType, 3>*> radiative_term,
+    const gsl::not_null<Cache*> cache,
+    detail::Tags::RadiativeTerm<DataType> /*meta*/) const {
+  get<0, 0>(*radiative_term) = 0.;
+  get<0, 1>(*radiative_term) = 0.;
+  get<0, 2>(*radiative_term) = 0.;
+  get<1, 1>(*radiative_term) = 0.;
+  get<1, 2>(*radiative_term) = 0.;
+  get<2, 2>(*radiative_term) = 0.;
```

Comment:
you can do: `std::fill(radiative_term->begin(), radiative_term->end(), 0.)`. Also in some more cases below.

Reply:
Done